pkg/azclient/auth.go

```diff
@@ -128,6 +128,20 @@ func NewAuthProvider(armConfig *ARMClientConfig, config *AzureAuthConfig, client
 		if err != nil {
 			return nil, err
 		}
+		if len(armConfig.NetworkResourceTenantID) > 0 && !strings.EqualFold(armConfig.NetworkResourceTenantID, armConfig.GetTenantID()) {
+			networkClientSecretCredential, err = azidentity.NewClientCertificateCredential(armConfig.NetworkResourceTenantID, config.GetAADClientID(), certificate, privateKey, credOptions)
```

Comment:
the logic lgtm, maybe it's better adding some logs to show it's cross tenant credential case?

Reply:
https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/internal/log/log.go is not exposed.

Reply:
you mean we cannot log info in this func?

Reply:
yes. the k8s logger may introduce k8s dependencies.